cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/util/FileUtils.java

```diff
@@ -0,0 +1,119 @@
+package cloud.foundry.cli.crosscutting.util;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import cloud.foundry.cli.crosscutting.exceptions.InvalidFileTypeException;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.hc.client5.http.HttpResponseException;
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.ProtocolException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This utility class provides a more easy to use interface for handling
+ * YAML files requested from a local file source or a remote host source.
+ *
+ */
+public class FileUtils {
+
+    private static final Set<String> ALLOWED_FILE_EXTENSIONS = new HashSet<>(Arrays.asList(
```

Comment:
Tbh YAML files _always_ carry the extension `.yml`, very rarely `.yaml`. I have never seen any files with upper-case extensions.
But otoh, it also doesn't hurt allowing `.yAmL` as extension.

Reply:
Before comparison, the file extension is converted to upper case. So `.yAml` is already supported.

Reply:
That's what I spotted below, and tried to express :)